ir/visitor.cpp

```diff
@@ -170,6 +175,17 @@ class Visitor::ChangeTracker {
         return it->second.result;
     }
 
+    /** Produce the final result of visiting @n.
+     *
+     * @return The ultimate result of visiting @n, or @n if `finish(@n)` has not
+     * been invoked.
+     */
+    const IR::Node *finalResult(const IR::Node *n) const {
```

Comment:
I think the interface could be a bit confusing, maybe it would be more straight-forward if it returned a final result, or nullptr if the final result does not exits?